### PR TITLE
Fix false positives for flex shorthand with 2 values in length-zero-no-unit

### PR DESCRIPTION
**Pull request by Sultanidze**
> Which issue, if any, is this issue related to?

#4831 

> Is there anything in the PR that needs further explanation?

e.g. "No, it's self-explanatory."

### CODE REVIEW

File: docs/user-guide/usage/node-api.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ As `config` contains no relative paths for `extends` or `plugins`, you do not ha
 ```js
 stylelint
   .lint({
-    config: { rules: "color-no-invalid-hex" },
+    config: { rules: { "color-no-invalid-hex": true } },
```
Comment (malsf21) [inline docs/user-guide/usage/node-api.md:83]: Is there a reason why you made this change? Doesn't seem relevant to the rule - was it an autoformat fix or something?

File: lib/rules/length-zero-no-unit/index.js [hunk 2]
```diff
@@ -40,6 +40,15 @@ function rule(actual, secondary, context) {
 			const ignorableIndexes = new Array(stringValue.length).fill(false);
 			const parsedValue = valueParser(stringValue);
 
+			if (decl.prop.toLowerCase() === 'flex') {
```
Comment (malsf21) [inline lib/rules/length-zero-no-unit/index.js:43]: May just want to add a comment pointing to the issue / MDN docs regarding the flex ambiguity - but will defer to others if they feel like this is appropriate!

**Review by malsf21 — COMMENTED** [review]
Looks like this has slipped through the cracks! On an initial glance, it looks good to me  - just left some minor feedback. Thanks for your work on this so far @Sultanidze!

(not sure if there is a more portable/elegant way of performing the check - I'll defer to the Stylelint pros 😄 )

File: docs/user-guide/usage/node-api.md (hunk 1 above)
Comment (Sultanidze) [reply docs/user-guide/usage/node-api.md:83]: Hi @malsf21 and thanks for review. I've encountered broken config in node api user guide (Node v12.19.0 is on my machine if it means).

File: lib/rules/length-zero-no-unit/index.js (hunk 2 above)
Comment (Sultanidze) [reply lib/rules/length-zero-no-unit/index.js:43]: @jeddy3 Should I add an inline comment here or issue number in commit message is enough?

Comment (jeddy3) [reply lib/rules/length-zero-no-unit/index.js:43]: An inline comment would be nice as this is tricky stuff. Perhaps:

```
// Allow <width> for last value in two-value and three-value flex syntax
// https://developer.mozilla.org/en-US/docs/Web/CSS/flex#syntax
```

**Review by jeddy3 — CHANGES_REQUESTED** [review]
@Sultanidze Apologies for the delay in reviewing this.

Can we add two tests with a comment, please? The logic looks good, but it's worth checking with a test. For example:

```
// accept
code: 'a { flEx: 1 /* comment */ 0px; }'

// reject
code: 'a { flex: 1 0 /* comment */ 0px; }', 
fixed: 'a { flex: 1 0 /* comment */ 0; }', 
```

**Comment by Sultanidze** [comment]
@jeddy3 thanks for advice to add test cases with comments. There are issues, I'm working on it.

**Comment by jeddy3** [comment]
Revisiting this now that the rule was refactored in https://github.com/stylelint/stylelint/pull/5256.

I think we can simply ignore the `flex` property to resolve the issue. Looking at [the syntax again](https://developer.mozilla.org/en-US/docs/Web/CSS/flex#syntax), this rule should ignore all possible values for the flex property as they'll either be a `<number>` or intentionally a `<width>`.

@Sultanidze Would you like to update the pull request?

I think a simple`isFlex(prop)` function on [line 108](https://github.com/stylelint/stylelint/blob/master/lib/rules/length-zero-no-unit/index.js#L108) should do it. And the following two `"accept"` tests should suffice:

```
a { flex: 1 0px; }
a { FLEX: 1 1 0px; }
```

No need to add anything to the description of the rule either.
